### ISSUE TITLE
Local scope and local locale

### ISSUE THREAD

**Issue by aplamon** [intlify/vue-i18n-next#915]
### Reporting a bug?

It seems with local scope the locale used is always the global, no matter what the local scope is set to with Composition API (but messages picked up are from local scope). 

### Expected behavior

Should use local locale for the component

### Reproduction

If I in component do:
const { t, locale } = useI18n({
    locale: "en-US",
    messages: {
        en: {
            VAT: 'VAT'
        },
        sv: {
            VAT: "Moms"
        }
    }
});
Even though I set locale to en-US here the component uses the global sv locale. It's not totally clear in the documentation, but it seems I should be able to have different locale for component by using local scope?

### System Info

```shell
System:
    OS: macOS 12.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 382.76 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 14.17.0 - ~/.nvm/versions/node/v14.17.0/bin/node
    Yarn: 1.22.11 - ~/.nvm/versions/node/v14.17.0/bin/yarn
    npm: 6.14.13 - ~/.nvm/versions/node/v14.17.0/bin/npm
  Browsers:
    Chrome: 97.0.4692.99
    Safari: 15.2
  npmPackages:
    vue-advanced-cropper: ^2.7.0 => 2.7.0
    vue-i18n: ^9.2.0-beta.30 => 9.2.0-beta.30
    vuex: ^4.0.1 => 4.0.2
```

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/intlify/vue-i18n-next/blob/master/.github/CONTRIBUTING.md)
- [X] Read the [Documentation](https://vue-i18n.intlify.dev/)
- [X] Check that there isn't [already an issue](https://github.com/intlify/vue-i18n-next/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/intlify/vue-i18n-next/discussions)
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by kazupon** [comment]
Thank you for your feedback!

You need to set `useScope: 'local'` on `useI18n`
docs said:
https://vue-i18n.intlify.dev/guide/advanced/composition.html#local-scope

Thanks!

**Comment by aplamon** [comment]
Thanks, I figured it out, I added that. I still think there is an issue here. I forgot to mention that I have it setup so that the keys are the default language ('en') and the local scope was not finding the key in 'en' so it some how reverted back to the root and whatever language it was created for. Once I added "translations" in the 'en' locale things started working as expected. So I guess that is the workaround. Although it is a little silly to have a 'en' file with entries such as "Save PDF": "Save PDF"
